### PR TITLE
Add tiff conversion for data endpoint

### PR DESCRIPTION
**Pull request by loichuder**

### CODE REVIEW

**Comment by t20100** [comment]
Why not adding `tiff` as a format to the [encode](https://github.com/silx-kit/h5grove/blob/main/h5grove/encoders.py#L87) function and `/data/` endpoint?

**Comment by andygotz** [comment]
> 

Do you mean as a protocol for sending images to the frontend?

**Comment by t20100** [comment]
@andygotz: No, there is already a binary transfer implemented for that, `tiff` is more for user to upload the data in a format they can read.

File: h5grove/encoders.py [hunk 1]
```diff
@@ -77,6 +78,12 @@ def npy_stream(array: Sequence[Number]) -> Generator[bytes, None, None]:
         yield chunk.tobytes("C")
 
 
+def convert_tiff(data: np.ndarray):
```
Comment (t20100) [inline h5grove/encoders.py:81]:
```suggestion
def convert_tiff(data: np.ndarray) -> bytes:
```

File: setup.cfg [hunk 2]
```diff
@@ -22,6 +22,7 @@ install_requires =
 	numpy
 	orjson
     h5py
+	tifffile
```
Comment (t20100) [inline setup.cfg:25]:
```suggestion
	h5py
	tifffile
```

Just to normalize `h5py` indentation

File: h5grove/encoders.py [hunk 3]
```diff
@@ -77,6 +78,12 @@ def npy_stream(array: Sequence[Number]) -> Generator[bytes, None, None]:
...
+    with io.BytesIO() as buffer:
+        tifffile.imwrite(buffer, data, photometric="minisblack")
```
Comment (t20100) [inline h5grove/encoders.py:83]: Did you checked what happens with special cases: strings/scalar, not 2D array, complex numbers?

**Review by t20100 — APPROVED** [review]
Some minor comments:
- `tiff` can be added to `format` param description in `docs/_static/api.yaml`
- `tiff` and `bin` can be added to `encode` docstring

LGTM

File: h5grove/encoders.py (hunk 3 above)
Comment (loichuder) [reply h5grove/encoders.py:83]: No but the same way `npy` is not meant be used with such cases, I don't expect the function to be called or to handle them.

If such a thing happens, I am happy to let the code raise an error by itself

Comment (t20100) [reply h5grove/encoders.py:83]: OK, sounds good.

**Pull request merged**